frame/core-fellowship/src/lib.rs

```diff
@@ -0,0 +1,593 @@
+			match T::InductOrigin::try_origin(origin) {
+				Ok(_) => {},
+				Err(origin) => ensure_root(origin)?,
+			}
+			ensure!(!Member::<T, I>::contains_key(&who), Error::<T, I>::AlreadyInducted);
+			ensure!(T::Members::rank_of(&who).is_none(), Error::<T, I>::Ranked);
+
+			T::Members::induct(&who)?;
+			let now = frame_system::Pallet::<T>::block_number();
+			Member::<T, I>::insert(
+				&who,
+				MemberStatus { is_active: true, last_promotion: now, last_proof: now },
+			);
+			Self::deposit_event(Event::<T, I>::Inducted { who });
+			Ok(Pays::No.into())
+		}
+
+		/// Increment the rank of a ranked and tracked account.
+		///
+		/// - `origin`: An origin which satisfies `PromoteOrigin` with a `Success` result of
+		///   `to_rank` or more or root.
+		/// - `who`: The account ID of the member to be promoted to `to_rank`.
+		/// - `to_rank`: One more than the current rank of `who`.
+		#[pallet::weight(T::WeightInfo::promote())]
+		#[pallet::call_index(5)]
+		pub fn promote(
+			origin: OriginFor<T>,
+			who: T::AccountId,
+			to_rank: RankOf<T, I>,
+		) -> DispatchResultWithPostInfo {
+			match T::PromoteOrigin::try_origin(origin) {
+				Ok(allow_rank) => ensure!(allow_rank >= to_rank, Error::<T, I>::NoPermission),
+				Err(origin) => ensure_root(origin)?,
+			}
+			let rank = T::Members::rank_of(&who).ok_or(Error::<T, I>::Unranked)?;
+			ensure!(
+				rank.checked_add(1).map_or(false, |i| i == to_rank),
+				Error::<T, I>::UnexpectedRank
+			);
+
+			let mut member = Member::<T, I>::get(&who).ok_or(Error::<T, I>::NotTracked)?;
+			let now = frame_system::Pallet::<T>::block_number();
+
+			let params = Params::<T, I>::get();
+			let rank_index = Self::rank_to_index(to_rank).ok_or(Error::<T, I>::InvalidRank)?;
+			let min_period = params.min_promotion_period[rank_index];
+			// Ensure enough time has passed.
+			ensure!(
+				member.last_promotion.saturating_add(min_period) <= now,
+				Error::<T, I>::TooSoon,
+			);
+
+			T::Members::promote(&who)?;
+			member.last_promotion = now;
+			member.last_proof = now;
+			Member::<T, I>::insert(&who, &member);
+			Self::dispose_evidence(who.clone(), rank, Some(to_rank));
+
+			Self::deposit_event(Event::<T, I>::Promoted { who, to_rank });
+
+			Ok(Pays::No.into())
+		}
+
+		/// Stop tracking a prior member who is now not a ranked member of the collective.
+		///
+		/// - `origin`: A `Signed` origin of an account.
+		/// - `who`: The ID of an account which was tracked in this pallet but which is now not a
+		///   ranked member of the collective.
+		#[pallet::weight(T::WeightInfo::offboard())]
+		#[pallet::call_index(6)]
+		pub fn offboard(origin: OriginFor<T>, who: T::AccountId) -> DispatchResultWithPostInfo {
+			let _ = ensure_signed(origin)?;
+			ensure!(T::Members::rank_of(&who).is_none(), Error::<T, I>::Ranked);
+			ensure!(Member::<T, I>::contains_key(&who), Error::<T, I>::NotTracked);
+			Member::<T, I>::remove(&who);
+			Self::deposit_event(Event::<T, I>::Offboarded { who });
```

Comment:
Also remove any leftover evidence? Or it inductively ensured that there is none?